### DISCUSSION TITLE
Configuring global functions

### DISCUSSION THREAD

**Discussion by Atulin** [eslint/eslint#14210] [Q&A] [answered]
I have a `site.js` file that's added with `<script src="/js/dist/site.min.js">` as one of the first scripts. It contains helper functions for `getCookieValue` and `setCookie`.

ESLint was complaining about undefined functions (`no-undef`), so I registered them as globals with

```json
    "globals": {
        "getCookieValue": "readonly",
        "setCookie": "readonly"
    },
```

The problem now, is that ESLint thinks I'm re-declaring those functions within the `site.js` file:

```
  62:10  error  'getCookieValue' is already defined as a built-in global variable  no-redeclare
  75:10  error  'setCookie' is already defined as a built-in global variable       no-redeclare
```

Any idea how I can tell it where the source of those functions is? Or should I just suppress the error in the source file?

**Comment by mdjermanovic** [comment, marked as answer]
Hi @Atulin!

I think the correct solution would be to set the globals to `"off"` for the file where they're declared, as they are not already declared before that script executes. 

That can be done either directly in the file, using the [`/* global */`](https://eslint.org/docs/user-guide/configuring/language-options#using-configuration-comments-1) configuration comment:

```js
/* global getCookieValue: off, setCookie: off */
```

or in the config file, using [`overrides`](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns):

```json
"globals": {
    "getCookieValue": "readonly",
    "setCookie": "readonly"
},
"overrides": [
    {
        "files": ["path/to/site.js"],
        "globals": {
            "getCookieValue": "off",
            "setCookie": "off"
        }
    }    
]
```

**Comment by Atulin** [comment]
That seems to have worked, thanks!

Feels like a bit of a hacky solution though. Perhaps something like

```json
"globals-sources": [
    "path/to/file.js"
]
```

that registers stuff from that file as globals and prevents it from being considered a re-declaration would be nice to have.
